### PR TITLE
Revisit #12019 : Show save dialog when calling a smart contract method and use the saved contact is already saved

### PR DESCRIPTION
**Pull request by segun**
Fixes: a bug introduced by this PR #12019 

Explanation:  
#12019 is a fix for the below:

Change the [domain] suggested gas fee (i) to Estimated gas fee (i)
Change that line's color from orange to black
Under that line, add Site suggested in italics in gray.

But it introduced the bug discussed [here](https://consensys.slack.com/archives/GTQAGKY5V/p1631220853285800)

`I'm seeing what I believe is a bug on v10.1.0. Creating transaction from within metamask shows the recipient as "New Contract", when it is one of the users own current accounts:
`

This is a fix for the bug.

Should not show add new address dialog under the below conditions
    Address is own adddress
    send to recipient header is not shown

Manual testing steps:  
  - 
  - 
  -

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -306,6 +306,7 @@
     "randomcolor": "^0.5.4",
     "rc": "^1.2.8",
     "react-devtools": "^4.11.0",
+    "react-test-renderer": "^17.0.2",
```
Comment (ryanml) [inline package.json:309]: is this actually being used? are we not able to accomplish something without this?

Comment (segun) [reply package.json]: No this is not needed. I was experimenting with it, but decided against it.

**Review by danjm — CHANGES_REQUESTED** [review]
This looks good! Nice work.

I think you just need to run `yarn setup` again, and then commit the lockfile changes, so that no lockfile changes are included from when you were experimenting with that other testing module.

**Review by danjm — APPROVED** [review]

**Review by brad-decker — APPROVED** [review]
LGTM

**Pull request merged**